src/tools/wallmap/objloader.go

```diff
@@ -0,0 +1,194 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"math"
+	"os"
+	"strings"
+)
+
+type Point2 struct {
+	X, Y float64
+}
+
+func (p *Point2) Scale(f float64) {
+	p.X = p.X * f
+	p.Y = p.Y * f
+}
+
+func Point2From3(p Point3) Point2 {
+	return Point2{p.X, p.Y}
+}
+
+type Point3 struct {
+	X, Y, Z float64
+}
+
+func (p *Point3) Set(s []string) error {
+	if _, err := fmt.Sscanf(s[0], "%f", &p.X); err != nil {
```

Comment:
`strconv.ParseFloat` is much faster; also `Sscanf` is always surprising with whitespace.